dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresVectorStore.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.VectorData;
+using Npgsql;
+
+namespace Microsoft.SemanticKernel.Connectors.Postgres;
+
+/// <summary>
+/// Represents a vector store implementation using PostgreSQL.
+/// </summary>
+public class PostgresVectorStore : IVectorStore
+{
+    private readonly IPostgresVectorStoreDbClient _postgresClient;
+    private readonly NpgsqlDataSource? _dataSource;
+    private readonly PostgresVectorStoreOptions? _options;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PostgresVectorStore"/> class.
+    /// </summary>
+    /// <param name="connectionString">Postgres database connection string.</param>
+    /// <param name="options">Optional configuration options for this class</param>
+    public PostgresVectorStore(string connectionString, PostgresVectorStoreOptions? options = default)
+    {
+        NpgsqlDataSourceBuilder dataSourceBuilder = new(connectionString);
+        dataSourceBuilder.UseVector();
+        this._dataSource = dataSourceBuilder.Build();
+        this._options = options ?? new PostgresVectorStoreOptions();
+        this._postgresClient = new PostgresVectorStoreDbClient(this._dataSource, this._options.Schema);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PostgresVectorStore"/> class.
+    /// </summary>
+    /// <param name="dataSource">Postgres data source.</param>
+    /// <param name="options">Optional configuration options for this class</param>
+    public PostgresVectorStore(NpgsqlDataSource dataSource, PostgresVectorStoreOptions? options = default)
+    {
+        this._dataSource = dataSource;
+        this._options = options ?? new PostgresVectorStoreOptions();
+        this._postgresClient = new PostgresVectorStoreDbClient(this._dataSource, this._options.Schema);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PostgresVectorStore"/> class.
+    /// </summary>
+    /// <param name="postgresDbClient">An instance of <see cref="IPostgresDbClient"/>.</param>
+    /// <param name="options">Optional configuration options for this class</param>
+    public PostgresVectorStore(IPostgresVectorStoreDbClient postgresDbClient, PostgresVectorStoreOptions? options = default)
+    {
+        this._postgresClient = postgresDbClient;
+        this._options = options ?? new PostgresVectorStoreOptions();
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<string> ListCollectionNamesAsync([EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        await foreach (string collection in this._postgresClient.GetTablesAsync(cancellationToken).ConfigureAwait(false))
+        {
+            yield return collection;
+        }
```

Comment:
Since the client already returns an `IAsyncEnumerable<string>`, you can remove the loop and just return directly.
You just need to remove the async from `ListCollectionNamesAsync` and the `[EnumeratorCancellation]` to make it work.